### PR TITLE
Update how TS is detected and loaded

### PR DESCRIPTION
**Pull request by alcuadrado**
This PR changes how TS is handled in Hardhat. Now, we consider the typescript mode to be enabled if the config is in ts. 

There's an edge case when testing plugins that use the `script-runner` module (i.e. `run('run', ...)`). Their fixture projects may have a `.js` config, so TS is not loaded in the subrprocess spawn by `script-runner`, and still try to import the plugin sources, which are in ts. This was only present in the ganache plugin, so I fixed it by turning their config files to .ts.

### CODE REVIEW

**Comment by alcuadrado** [comment]
Oh, I based this PR in `plugins-cleanup`, as that one changes how ts is handled in other parts of `hardhat-core`.

File: packages/hardhat-core/src/internal/core/typescript-support.ts [hunk 1]
```diff
@@ -1,9 +1,28 @@
...
+ */
+export function willRunWithTypescript(configPath?: string): boolean {
+  const config = resolveConfigPath(configPath);
+  return isTypescriptFile(config);
+}
+
+/**
+ * Returns true if an Hardhat is already running with typescript.
+ */
+export function isRunningWithTypescript(config: HardhatConfig): boolean {
```
Comment (fvictorio) [inline packages/hardhat-core/src/internal/core/typescript-support.ts:22]: This function seems to only be used in the tests

Comment (alcuadrado) [reply packages/hardhat-core/src/internal/core/typescript-support.ts:22]: It's used in the `test` task. I'm not sure if it should receive the config tbh. Maybe the HRE is better?

Comment (fvictorio) [reply packages/hardhat-core/src/internal/core/typescript-support.ts:22]: Oh, my bad, I read "test" in the list of files that referenced it and didn't notice it was the task :sweat_smile: 

> I'm not sure if it should receive the config tbh. Maybe the HRE is better?

I think the config is fine. In general, passing the HRE feels like _too much_ and I think we should avoid it (I don't have a strong argument, it just feels wrong to pass an object that can do a lot of things).

**Review by fvictorio — APPROVED** [review]

**Pull request merged**